zcnbridge/config.go

```diff
@@ -1,260 +1,97 @@
-				EthereumNodeURL: cfg.EthereumNodeURL,
-				GasLimit:        cfg.GasLimit,
-				Value:           cfg.Value,
-			},
-			EthereumAddress: cfg.EthereumAddress,
-			Password:        cfg.Password,
-			Homedir:         ".",
-		},
-		BridgeConfig: &BridgeConfig{
-			ConsensusThreshold: cfg.ConsensusThreshold,
-		},
-		Instance: &Instance{
-			startTime: common.Now(),
-			zcnWallet: wallet,
-		},
+		BridgeAddress:      cfg.GetString("bridge.bridge_address"),
+		TokenAddress:       cfg.GetString("bridge.token_address"),
+		AuthorizersAddress: cfg.GetString("bridge.authorizers_address"),
+		EthereumAddress:    cfg.GetString("bridge.ethereum_address"),
+		Password:           cfg.GetString("bridge.password"),
+		EthereumNodeURL:    cfg.GetString("ethereum_node_url"),
+		GasLimit:           cfg.GetUint64("bridge.gas_limit"),
+		ConsensusThreshold: cfg.GetFloat64("bridge.consensus_threshold"),
+		Homedir:            homedir,
 	}
 }
 
-func (b *BridgeClient) ClientID() string {
-	return b.zcnWallet.ClientID
-}
-
-func (b *BridgeOwner) ClientID() string {
-	return b.zcnWallet.ClientID
-}
-
 // SetupBridgeClientSDK Use this from standalone application
 // 0Chain SDK initialization is required
 func SetupBridgeClientSDK(cfg *BridgeSDKConfig, walletFile ...string) *BridgeClient {
 	log.InitLogging(*cfg.Development, *cfg.LogPath, *cfg.LogLevel)
-	bridgeClient := CreateBridgeClient(initBridgeConfig(cfg), walletFile...)
+	bridgeClient := CreateBridgeClient(initChainConfig(cfg), walletFile...)
 	return bridgeClient
 }
 
-// SetupBridgeOwnerSDK Use this from standalone application to initialize bridge owner.
-// 0Chain SDK initialization is not required in this case
-func SetupBridgeOwnerSDK(cfg *BridgeSDKConfig, walletFile ...string) *BridgeOwner {
-	log.InitLogging(*cfg.Development, *cfg.LogPath, *cfg.LogLevel)
-	bridgeOwner := CreateBridgeOwner(initBridgeConfig(cfg), walletFile...)
-	return bridgeOwner
-}
-
-func loadWallet(homedir string, fileName ...string) (*zcncrypto.Wallet, error) {
-	var walletPath string
+func initChainConfig(sdkConfig *BridgeSDKConfig) *viper.Viper {
+	cfg := readConfig(sdkConfig, func() string {
+		return *sdkConfig.ConfigChainFile
+	})
 
-	if len(fileName) != 0 {
-		walletPath = path.Join(homedir, fileName[0])
-	} else {
-		walletPath = path.Join(homedir, ZChainWalletConfigName)
-	}
+	log.Logger.Info(fmt.Sprintf("Chain config has been initialized from %s", cfg.ConfigFileUsed()))
 
-	clientBytes, err := os.ReadFile(walletPath)
-	if err != nil {
-		ExitWithError("Error reading the wallet", err)
-	}
+	return cfg
+}
 
-	wallet := &zcncrypto.Wallet{}
-	err = json.Unmarshal(clientBytes, &wallet)
+func readConfig(sdkConfig *BridgeSDKConfig, getConfigName func() string) *viper.Viper {
+	cfg := viper.New()
+	cfg.AddConfigPath(*sdkConfig.ConfigDir)
+	cfg.SetConfigName(getConfigName())
+	cfg.SetConfigType("yaml")
+	err := cfg.ReadInConfig()
 	if err != nil {
-		return nil, err
+		_, file, line, ok := runtime.Caller(2)
```

Comment:
do we really need to return the file name and line number when the error happens. Knowing the err should be enough, if we really need to know where the  error happens, give more info in the error message to differ it from common error should be enough as well.

Reply:
Done